contracts/oracles/TwapOracle.sol

```diff
@@ -196,30 +192,47 @@ contract TwapOracle is OwnableUpgradeable, TwapInterface {
     }
 
     /**
-     * @notice Update new and old observations of lagging window if period elapsed.
+     * @notice Append current Observation and pick equal or just greater than window start timestamp,
```

Comment:
```suggestion
     * @notice Append current observation and pick equal or just greater than window start timestamp.
```

Reply:
Yes Corey, this is actually correct. In the comment, we are talking about the else condition. Updated the comment to consider the situation inside if condition as well.

Reply:
The windowStartTimestamp is stored in both cases.